nengo/config.py

```diff
@@ -62,10 +66,32 @@ def __setattr__(self, key, value):
                 raise ConfigError("Parameter '%s' is not configurable" % key)
             param.set_default(self, value)
 
+    def __getstate__(self):
+        state = {k: getattr(self, k) for k in (
+            '_configures', '_extraparams', '_defaultparams')}
+
+        # Store all of the things we set in the params
+        for attr in self.params():
+            param = self.get_param(attr)
+            if self in param:
+                state[attr] = param.get_default(self)
+
+        return state
+
+    def __setstate__(self, state):
+        self._configures = state['_configures']
+        self._defaultparams = tuple(state['_defaultparams'])
+        self._extraparams = dict(state['_extraparams'])
```

Comment:
We may not have to... I copied this from my old branch that implemented partial copies. I can take them out and see if it matters